src/core/zowe/core_for_zowe_sdk/sdk_api.py

```diff
@@ -11,48 +11,44 @@
 """
 from .request_handler import RequestHandler
 from .constants import constants
-from .connection import ApiConnection
-from .zosmf_profile import ZosmfProfile
+from .session import Session, ISession
+from .zosmf_profile2 import ProfileManager
+from . import session_constants
 
 
 class SdkApi:
     """
     Abstract class used to represent the base SDK API.
-
-    Attributes
-    ----------
-    connection: dict
-        A dictionary containing the connection arguments
-    default_url: str
-        The default endpoint for the API
     """
 
-    def __init__(self, connection, default_url):
-        if "plugin_profile" in connection:
-            self.connection = ZosmfProfile(connection['plugin_profile']).load()
-        else:
-            self.connection = ApiConnection(**connection)
+    def __init__(self, default_url, session):
+        self.session: ISession = session
 
-        self.constants = constants
         self.default_service_url = default_url
         self.default_headers = {
             "Content-type": "application/json",
-            "X-CSRF-ZOSMF-HEADER": ""
+            "X-CSRF-ZOSMF-HEADER": "",
         }
+
         self.request_endpoint = "https://{base_url}{service}".format(
-            base_url=self.connection.host_url, service=self.default_service_url
+            base_url=Session.host_url, service=self.default_service_url
```

Comment:
There's a `Session.host_url` property does contains that. If the session class is imported into the sdkapi, it should work right?

Reply:
Yes, the `Session.host_url` property contains the information we want 🙂 Since it has an `@property` decorator, we need to access the `host_url` of a session instance, not the `host_url` of the static Session class.

If you store an instance of the Session class earlier:
```
        session = Session(profile)
        self.session: ISession = session.load()
```

Then you should be able to generate the request endpoint like this:
```
        self.request_endpoint = session.host_url + self.default_service_url
```

Reply:
Understood

Reply:
The `request_endpoint` is still not generated correctly because the protocol is being duplicated.

I would suggest changing it to use the code in the comment above. The advantage of doing it this way is:
```
session = Session(profile)
self.session: ISession = session.load()
```
lets us store both the Session class and the ISession dataclass for future use.

And then we can make use of the `Session.host_url` property without duplicating code for building the host URL string:
```
self.request_endpoint = session.host_url + self.default_service_url
```